DiscordTools/DiscordClient.cs

```diff
@@ -114,6 +114,34 @@ public void Close()
       IsConnected = false;
     }
 
+    public void UploadArtwork(string artworkData)
+    {
+      // Clone the string
+      var _tmpHash = new StringBuilder(_currentArtworkHash).ToString();
+
+      if (IsConnected)
+      {
+        if (_assetManager?.initialised == true)
+        {
+          var assetId = _assetManager.GetCachedAsset(artworkData);
+          if (assetId == null)
+          {
+            assetId = AssetManager.ASSET_LOGO;
+            _ = _assetManager.UploadAsset(artworkData).ContinueWith(
+              t =>
+              {
+                var id = t.Result;
+                if (id == null)
+                { // this most likely means another task is already uploading this asset.
+                return;
+                }
```

Comment:
Why did you remove the presence update?
Maybe I am missing something here but if the artwork matches the current track (can be checked using `_currentArtworkHash`) it should update the presence if it finished uploading.

Reply:
Nah you're right, it's barely any work, and we can deduplicate the code.

Reply:
Wait no, I remembered why I didn't do this:
https://github.com/sll552/DiscordBee/blob/master/DiscordTools/DiscordClient.cs#L119
`RichPresence` is cloned at call time. Unless I'm mistaken, if we se this in the `ContinueWith` with just `LargeImageKey` changed, it would set the correct image, but everything else would be wrong (whatever it was when the image was requested to be preloaded).

Reply:
I'm not sure why but this doesn't seem to be a problem actually. I'm not sure why but then again I didn't test that much.

Reply:
Did more testing, seems fine, can probably merge. Pushing now.

Reply:
Ohh I see why it's not an issue. `_discordPresence` is a class member 🙃 

Imo it's a bit confusing that the underscore naming convention is used both for class members, but also for locals in some places.

Reply:
This was a bit annoying to do right when I tried so I stopped working on it, idk if/when I'll pick it up.

Reply:
Please test my take on this, seemed working for me but with the caching being so inconsistent it might as well be by pure luck.